src/spec-lang/tc/typecheck.ts

```diff
@@ -1500,6 +1514,66 @@ function matchArguments(
     return true;
 }
 
+/**
+ * We check the following in forall (uint t in [a ... b]) e(t):
+ *   - Type of t should be numeric
+ *   - a and b should be numeric and be castable to the type of t
+ *   - type of e is boolean
+ *   - return value of this expression is boolean
+ *   - t is defined in e(t).
+ */
+export function tcForAll(expr: SForAll, ctx: STypingCtx, typeEnv: TypeEnv): TypeNode {
+    assert(
+        typeof expr.start == typeof expr.end,
```

Comment:
This should be `expr._start` and `expr._end`. `expr.start` and `expr.end` are just functions - they are always defined.

Reply:
We can remove this assert now right?

Reply:
Yes. Though it would be good to add a special-case error message here for the case `forall (type i in a)` for when `a` is not an array, and for when `type` is not a uint. Otherwise the error messages may be confusing, as they will refer to `a.length` which not present in the code. This is not a blocker for this PR, should be fixed in the next

Reply:
Fixed.